### PR TITLE
#1309 Revamp testing docs

### PR DESCRIPTION
**Pull request by zgavin1**
@Andarist 

Here's a first attempt, I'm super open if you or any others have more suggestions to add or refine this.

#1309

### CODE REVIEW

**Comment by zgavin1** [comment]
I'll see what happened with the build when I have a moment this afternoon.

File: docs/advanced/Testing.md [hunk 1]
```diff
@@ -220,3 +220,256 @@ See also: Repository Examples:
...
+  // assert that your saga does nothing unexpected
+  assert.true(actualEffects.length === 2);
+});
+```
+
+### redux-saga-tester
+
+A final library to consider for integration testing. this library provides a `sagaTester` class, to which you instantiate with your store's initial state and your reducer.
+
+To test your saga, the `sagaTester` instance `start()` method with your saga and it's argument(s). This runs your saga to it's end, and then you may assert that effects occured, actions were dispatched and the state was updated as expected.
```
Comment (Andarist) [inline docs/advanced/Testing.md]: should be `its` instead of `it's`

**Comment by Andarist** [comment]
This looks great 😍 Don't think there is anything to add right now, we can always work on it later too. I've noticed only a single small grammar mistake.

**Comment by zgavin1** [comment]
Cool! Thanks for reviewing it- I'll make that change and find the build issue later tonight.

**Comment by Andarist** [comment]
Build issue is not related to your changes, I won't block this PR because of it.

**Comment by Andarist** [comment]
Thanks, this is AWESOME 🎉

**Pull request merged**